string grammar
string[] valid_gram_id
---
bool success
